static/src/javascripts/projects/common/modules/commercial/adblock-ask.js

```diff
@@ -25,35 +29,19 @@ const askHtml = `
 </div>
 `;
 
-export const adblockTest: ABTest = {
-    id: 'AdblockAsk',
-    start: '2019-02-20',
-    expiry: '2021-02-16',
-    author: 'Tom Forbes',
-    description:
-        'Places a contributions ask underneath the right-hand ad slot on articles.',
-    audience: 1,
-    audienceOffset: 0,
-    successMeasure: '',
-    audienceCriteria: '',
-    showForSensitive: true,
-    canRun() {
-        return (
-            !shouldHideSupportMessaging() &&
-            !pageShouldHideReaderRevenue() &&
-            !config.get('page.hasShowcaseMainElement')
-        );
-    },
+const canShow = () =>
+    !shouldHideSupportMessaging() &&
+    !pageShouldHideReaderRevenue() &&
+    !config.get('page.hasShowcaseMainElement');
 
-    variants: [
-        {
-            id: 'control',
-            test: (): void => {
-                const slot = document.querySelector('.js-aside-slot-container');
-                if (slot) {
-                    slot.innerHTML += askHtml;
-                }
-            },
-        },
-    ],
+export const initAdblockAsk = () => {
+    fastdom
+        .read(() => $('.js-aside-slot-container'))
+        .then(slot => {
+            if (canShow()) {
```

Comment:
This is nitpicking a bit, but you could call canShow first, before fastdom.read, which would avoid doing any DOM reads if canShow is false.

Reply:
Good point, thank Jon.